pkg/util/arangod/client.go

```diff
@@ -156,20 +156,38 @@ func CreateArangodImageIDClient(ctx context.Context, deployment k8sutil.APIObjec
 
 // CreateArangodClientForDNSName creates a go-driver client for a given DNS name.
 func createArangodClientForDNSName(ctx context.Context, cli corev1.CoreV1Interface, apiObject *api.ArangoDeployment, dnsName string) (driver.Client, error) {
-	connConfig, err := createArangodHTTPConfigForDNSNames(ctx, cli, apiObject, []string{dnsName})
-	if err != nil {
-		return nil, maskAny(err)
-	}
-	// TODO deal with TLS with proper CA checking
-	conn, err := http.NewConnection(connConfig)
-	if err != nil {
-		return nil, maskAny(err)
+	config := driver.ClientConfig{}
+	vst := driver.mustUseVST(context)
+	if vst {
+		connConfig, err := createArangodVSTConfigForDNSNames(ctx, cli, apiObject, []string{dnsName})
+		if err != nil {
+			return nil, maskAny(err)
+		}
+		// TODO deal with TLS with proper CA checking
+		conn, err := vst.NewConnection(connConfig)
+		if err != nil {
+			return nil, maskAny(err)
+		}
+		// Create client
```

Comment:
From this point on the code paths are the same. Only create `conn` in the if/else block.
Use `var conn driver.Connection` before the if/else block.
Then change `conn, err := ...` to `conn, err = ...`